doc/dev/msgr2.rst

```diff
@@ -288,20 +291,17 @@ an established session.
     __le64 flags (CEPH_MSG_CONNECT_* bitmask)
     __le64 cookie
 
-  - The cookie can be used by the client if it is later disconnected and wants to
-    reconnect and resume the session.
+  - The server cookie can be used by the client if it is later disconnected
+    and wants to reconnect and resume the session.
 
 * TAG_RECONNECT (client->server): reconnect to an established session::
 
     __le32 num_addrs
     entity_addr_t * num_addrs
-    entity_addr_t target entity addr
-    __le64 cookie
-    __le64 id (of name.id, e.g., osd.123 -> 123)
+    __le64 client_cookie
+    __le64 server_cookie
     __le64 global_seq
     __le64 connect_seq
-    __le64 supported_features
-    __le64 required_features
```

Comment:
AFAICS we still need the id and features:

- connect a -> b
- a sends client_ident
- fault (b has id and features, but a know who b is)
- connect b -> a
- reconnect
- fault(full=false)
- reconnect

...and a still thinks b is unknown.0 without features.

```
   -20> 2019-02-18 18:02:02.840 7ffaa5540700  1 -- [v2:172.21.15.119:3300/0,v1:172.21.15.119:6789/0] <== unknown.0 v2:172.21.15.144:3300/0 32 ==== paxos(accept lc 8 fc 0 pn 300 opn 0) v4 ==== 84+0+0 (1802574957 0 0) 0x55d4de6a6b00 con 0x55d4dd612c00
```
from /a/sage-2019-02-18_17:40:49-rados:monthrash-wip-sage2-testing-2019-02-18-0948-distro-basic-smithi/3607676

Reply:
Found the problem that is fixed with https://github.com/ceph/ceph/pull/26484/commits/c944f6a9cb724e475fe92c489e34048610384c51

Reply:
ups, need to amend, features are also getting smashed

Reply:
If peer A is reconnecting to peer B, B should already know the features and entity_name, but we were smashing that info in `reuse_connection` method. https://github.com/ceph/ceph/pull/26484/commits/52dc478301045c96651208f95ebacf1dcbfa737c fixed that by only setting that info when `reuse_connection` is called from `handle_client_ident`.